### ISSUE TITLE
Remove version 5 requirement for styled-components

### ISSUE THREAD

**Issue by hrvstr** [waspeer/sanity-plugin-iconify#4]
The current version requirement for styled-components in the sanity-plugin-iconify package's peerDependencies is ^5. However, this conflicts with other dependencies that require version ^6.1.8, causing installation issues and dependency conflicts.

I kindly request that the version requirement for styled-components be updated to ^6 or a higher compatible version to resolve these conflicts and ensure smoother installations for users.

Installing with `--force` seems to work and I didn't encounter any issues. Any reason you keep the version fixed to v5?

**Comment by waspeer** [comment]
Hi! Thanks for filing an issue.

This version constraint is part of the [default plugin scaffolding provided by Sanity](https://github.com/sanity-io/plugin-kit). Looking through the code it seems like they still [enforce version `^5.2`](https://github.com/sanity-io/plugin-kit/blob/7741cf4452bd36b7adabf71d94bce4f7f6c07818/src/configs/forced-package-versions.ts#L12) for styled components. 

I'm happy to change this requirement to `^6` but I want to ensure this will not lead to unexpected errors, eg with the studio.

Did you run into this issue while working in a newly initialized instance of Sanity Studio?

**Comment by hrvstr** [comment]
I updated an existing Sanity instance with `ncu` and came across this version conflict. I see they require ^5.2 in their plugin kit but the newest version of Sanity itself seems to [require ^6.1.6](https://github.com/sanity-io/sanity/blob/c9ceac00389fe04f3e599d9f331aeff0dbf279a8/packages/sanity/package.json#L335) which seems weird to me. Maybe I can ask Sanity support in the Slack channel about this issue.

I just tried using an icon field and everything seems to work as expected after installing with `--force`.

```
{
  "name": "xyz",
  "private": true,
  "version": "1.0.0",
  "main": "package.json",
  "license": "UNLICENSED",
  "scripts": {
    "dev": "sanity dev",
    "start": "sanity start",
    "build": "sanity build",
    "deploy": "sanity deploy",
    "deploy-graphql": "sanity graphql deploy",
    "edit": "node scripts/editDocument.js"
  },
  "keywords": [
    "sanity"
  ],
  "dependencies": {
    "@sanity/color-input": "^3.1.1",
    "@sanity/vision": "^3.33.0",
    "phone": "^3.1.42",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-icons": "^5.0.1",
    "react-is": "^18.2.0",
    "sanity": "^3.33.0",
    "sanity-pills": "^2.0.0",
    "sanity-plugin-iconify": "^1.1.1",
    "sanity-plugin-media": "^2.2.5",
    "sanity-plugin-netlify": "^1.1.2",
    "styled-components": "^6.1.8"
  },
  "devDependencies": {
    "@sanity/eslint-config-studio": "^3.0.1",
    "@types/react": "^18.2.65",
    "@types/styled-components": "^5.1.34",
    "eslint": "^8.57.0",
    "prettier": "^3.2.5",
    "typescript": "^5.4.2"
  },
  "prettier": {
    "semi": false,
    "printWidth": 100,
    "bracketSpacing": false,
    "singleQuote": true
  }
}

```

**Comment by waspeer** [comment]
Did you get any clarity from the Sanity slack channel?
Maybe I could update the requirement to `^5 || ^6.1`? If Sanity itself works with version `6.1` I think that might be fine. Wdyt?

**Comment by hrvstr** [comment]
I created a post in the #help channel but didn't get a reply so far:
https://sanity-io-land.slack.com/archives/C9Z7RC3V1/p1711100373923919

I did some more testing and noticed that Sanity sets the version to 6.1.8 on a fresh install.

```
❯ npm list | grep styled-components
├── styled-components@6.1.8
```

The `npm create sanity@latest` script also uses the `--legacy-peer-deps` argument which makes installing the latest studio in combination with this plugin possible. I don't know if this is the standard for sanity but I think it should be safe to set the requirement to `^5 || ^6.1`.
